### PR TITLE
feat: Added mq configurations revisions

### LINKED ISSUE #516: MQ Broker configuration revisions are not fetched
### Describe the resource.

We might want to fetch `aws_mq_broker_configuration_revisions`, which contains all the revisions of all mq broker configurations. Currently we have `aws_mq_broker_configurations`, which contain the latest revision.

### Use Case

It needs to be a new resource because the output format of `DescribeConfigurationRevisionOutput` differs from `DescribeConfigurationOutput`, mainly the former has a base64-encoded XML payload as config while the latter has each field separately.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amanenk**
- removed redundant fields from relations
- renamed `broker_id` to `id`
- removed redundant primary keys
- revision data base64 is decoded and converted from xml to json and looks next:
```
{
    "broker": {
        "-xmlns": "http://activemq.apache.org/schema/core",
        "-schedulePeriodForDestinationPurge": "10000",
        "destinationPolicy": {
            "policyMap": {
                "policyEntries": {
                    "policyEntry": [
                        {
                            "-gcInactiveDestinations": "true",
                            "-inactiveTimoutBeforeGC": "600000",
                            "pendingMessageLimitStrategy": {
                                "constantPendingMessageLimitStrategy": {
                                    "-limit": "1000"
                                }
                            },
                            "-topic": "\u003e"
                        },
                        {
                            "-inactiveTimoutBeforeGC": "600000",
                            "-queue": "\u003e",
                            "-gcInactiveDestinations": "true"
                        }
                    ]
                }
            }
        },
        "plugins": ""
    }
}
```

closes #516

### CODE REVIEW

File: resources/services/mq/brokers.go [hunk 1]
```diff
@@ -274,31 +265,53 @@ func MqBrokers() *schema.Table {
 						Type:        schema.TypeJSON,
 					},
 				},
+				Relations: []*schema.Table{
+					{
+						Name:          "aws_mq_broker_configuration_revisions",
+						Resolver:      fetchMqBrokerConfigurationRevisions,
+						IgnoreInTests: true,
+						Columns: []schema.Column{
```
Comment (disq) [inline resources/services/mq/brokers.go:273]: missing PK? probably broker_configuration_cq_id, configuration_id?

File: resources/services/mq/brokers.go [hunk 2]
```diff
@@ -274,31 +265,53 @@ func MqBrokers() *schema.Table {
...
+					},
+				},
 			},
 			{
-				Name:        "aws_mq_broker_users",
-				Resolver:    fetchMqBrokerUsers,
-				IgnoreError: client.IgnoreAccessDeniedServiceDisabled,
-				Options:     schema.TableCreationOptions{PrimaryKeys: []string{"broker_cq_id", "username"}},
+				Name:     "aws_mq_broker_users",
+				Resolver: fetchMqBrokerUsers,
```
Comment (disq) [inline resources/services/mq/brokers.go:307]: PK removed?

Comment (amanenk) [reply resources/services/mq/brokers.go:307]: as we don't use manual PKs on relations I have removed them. They are created automatically if not set

File: resources/services/mq/brokers.go [hunk 3]
```diff
@@ -274,31 +265,53 @@ func MqBrokers() *schema.Table {
...
-						Description: "The AWS Account ID of the resource.",
-						Type:        schema.TypeString,
-						Resolver:    client.ResolveAWSAccount,
-					},
-					{
-						Name:        "region",
-						Description: "The AWS Region of the resource.",
-						Type:        schema.TypeString,
-						Resolver:    client.ResolveAWSRegion,
-					},
```
Comment (roneli) [inline resources/services/mq/brokers.go:301]: wouldn't remove account_id, region

File: resources/services/mq/brokers.go [hunk 4]
```diff
@@ -441,6 +465,57 @@ func fetchMqBrokerConfigurations(ctx context.Context, meta schema.ClientMeta, pa
...
+
+	input := mq.ListConfigurationRevisionsInput{ConfigurationId: cfg.Id}
+	for {
+		output, err := svc.ListConfigurationRevisions(ctx, &input, func(options *mq.Options) {
+			options.Region = c.Region
+		})
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		for _, rev := range output.Revisions {
```
Comment (roneli) [inline resources/services/mq/brokers.go:484]: nit: would maybe describe all revisions, then pass them to the channel

File: resources/services/mq/brokers.go [hunk 5]
```diff
@@ -274,31 +265,53 @@ func MqBrokers() *schema.Table {
 						Type:        schema.TypeJSON,
 					},
 				},
+				Relations: []*schema.Table{
+					{
+						Name:          "aws_mq_broker_configuration_revisions",
+						Resolver:      fetchMqBrokerConfigurationRevisions,
+						IgnoreInTests: true,
```
Comment (roneli) [inline resources/services/mq/brokers.go:272]: can we add terraforms to test this? (we can do it in a separate PR, merge it and then tests will pass here and we can merge this)

File: resources/services/mq/brokers.go [hunk 6]
```diff
@@ -197,18 +201,6 @@ func MqBrokers() *schema.Table {
...
-						Description: "The AWS Account ID of the resource.",
-						Type:        schema.TypeString,
-						Resolver:    client.ResolveAWSAccount,
-					},
-					{
-						Name:        "region",
-						Description: "The AWS Region of the resource.",
-						Type:        schema.TypeString,
-						Resolver:    client.ResolveAWSRegion,
-					},
```
Comment (roneli) [inline resources/services/mq/brokers.go:211]: why remove region/account_id ?

File: resources/services/mq/brokers.go [hunk 7]
```diff
@@ -45,29 +50,30 @@ func MqBrokers() *schema.Table {
...
 			{
 				Name:        "arn",
-				Description: "The Amazon Resource Name (ARN) of the broker.",
+				Description: "The broker's Amazon Resource Name (ARN).",
 				Type:        schema.TypeString,
 				Resolver:    schema.PathResolver("BrokerArn"),
 			},
 			{
-				Name:        "broker_id",
+				Name:        "id",
```
Comment (roneli) [inline resources/services/mq/brokers.go:58]: this is a breaking change, can you verify that know policies uses this field?

**Review by roneli — CHANGES_REQUESTED** [review]
Add some review comments, would double check the migrations with existing data.

**Comment by roneli** [comment]
would update migrations to v0.10.16 (just so this PR will be 1 above the rest) so it will be inserted and we tag so less conflicts.

File: resources/services/mq/brokers.go (hunk 7 above)
Comment (amanenk) [reply resources/services/mq/brokers.go:58]: no policies use this column

File: resources/provider/migrations/postgres/22_v0.10.14.up.sql [hunk 8]
```diff
@@ -1,38 +1,43 @@
 -- Resource: ec2.instance_statuses
-CREATE TABLE IF NOT EXISTS "aws_ec2_instance_statuses" (
```
Comment (disq) [inline resources/provider/migrations/postgres/22_v0.10.14.up.sql:2]: Should not edit migration files from previously tagged releases, this causes confusion down the line. Please revert this change.

File: resources/provider/migrations/timescale/22_v0.10.14.up.sql [hunk 9]
```diff
@@ -1,42 +1,48 @@
 -- Resource: ec2.instance_statuses
-CREATE TABLE IF NOT EXISTS "aws_ec2_instance_statuses" (
```
Comment (disq) [inline resources/provider/migrations/timescale/22_v0.10.14.up.sql:2]: Should not edit migration files from previously tagged releases, this causes confusion down the line. Please revert this change.

**Review by disq — DISMISSED** [review]

File: resources/services/mq/brokers.go (hunk 5 above)
Comment (amanenk) [reply resources/services/mq/brokers.go:272]: objects are already there. no need in terraform

File: resources/provider/migrations/timescale/22_v0.10.14.up.sql (hunk 9 above)
Comment (amanenk) [reply resources/provider/migrations/timescale/22_v0.10.14.up.sql:2]: only formatting has changed

File: resources/provider/migrations/postgres/22_v0.10.14.up.sql (hunk 8 above)
Comment (amanenk) [reply resources/provider/migrations/postgres/22_v0.10.14.up.sql:2]: only formatting has changed

File: resources/provider/migrations/timescale/22_v0.10.14.up.sql (hunk 9 above)
Comment (disq) [reply resources/provider/migrations/timescale/22_v0.10.14.up.sql:2]: I know but it still causes confusion when we're inspecting releases and fixing migrations.

File: resources/provider/migrations/postgres/22_v0.10.14.up.sql (hunk 8 above)
Comment (roneli) [reply resources/provider/migrations/postgres/22_v0.10.14.up.sql:2]: would maybe discard any changes to this file

File: resources/provider/migrations/timescale/22_v0.10.14.up.sql (hunk 9 above)
Comment (roneli) [reply resources/provider/migrations/timescale/22_v0.10.14.up.sql:2]: would remove changes for this file

File: resources/provider/migrations/timescale/24_v0.10.16.up.sql [hunk 10]
```diff
@@ -4,3 +4,37 @@
...
+ALTER TABLE IF EXISTS aws_mq_brokers
+    DROP CONSTRAINT aws_mq_brokers_pk;
+ALTER TABLE IF EXISTS "aws_mq_brokers"
+    RENAME COLUMN broker_id TO id;
+ALTER TABLE IF EXISTS aws_mq_brokers
+    ADD CONSTRAINT aws_mq_brokers_pk PRIMARY KEY (cq_fetch_date, account_id, id);
+ALTER TABLE IF EXISTS aws_mq_broker_configurations
+    DROP CONSTRAINT aws_mq_broker_configurations_pk;
+ALTER TABLE IF EXISTS aws_mq_broker_configurations
+    ADD CONSTRAINT aws_mq_broker_configurations_pk PRIMARY KEY (cq_fetch_date, cq_id);
```
Comment (roneli) [inline resources/provider/migrations/timescale/24_v0.10.16.up.sql:18]: I am not sure if dropping a PK on a hypertable and recreating will work out of the box. I would double check that with a TSDB flavor db.

**Review by roneli — APPROVED** [review]
LGTM, approving as I have only 2 nits (remove changes for old migrations files) + double check that the TSDB migration works (pk dropping and recreating on hypertables)

**Review by roneli — APPROVED** [review]
